src/main.rs

```diff
@@ -588,32 +589,38 @@ struct TestFile {
     shard_post_state: TestShardState,
 }
 
-impl From<TestBeaconState> for BeaconState {
-    fn from(input: TestBeaconState) -> Self {
-        BeaconState {
+impl TryFrom<TestBeaconState> for BeaconState {
+    type Error = &'static str;
+
+    fn try_from(input: TestBeaconState) -> Result<Self, Self::Error> {
+        Ok(BeaconState {
             execution_scripts: input
                 .execution_scripts
                 .iter()
                 .map(|x| ExecutionScript { code: load_file(x) })
                 .collect(),
-        }
+        })
     }
 }
 
-impl From<TestShardBlock> for ShardBlock {
-    fn from(input: TestShardBlock) -> Self {
-        ShardBlock {
+impl TryFrom<TestShardBlock> for ShardBlock {
+    type Error = &'static str;
+
+    fn try_from(input: TestShardBlock) -> Result<Self, Self::Error> {
+        Ok(ShardBlock {
             env: input.env,
             data: ShardBlockBody {
                 data: input.data.from_hex().expect("invalid hex data"),
```

Comment:
Need to change this to the question mark operator, but means we need to introduce our own error type (since can't convert `FromHexError` to `&str`).